test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -713,51 +715,39 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 
 			By("Verifying NRTs had no updates because the pods failed to be scheduled on any node")
 			e2efixture.MustSettleNRT(fxt)
-			_, err = wait.With(fxt.Client).Interval(5*time.Second).Timeout(1*time.Minute).ForNodeResourceTopologiesEqualTo(context.TODO(), &nrtInitialList, func(nrt *nrtv1alpha2.NodeResourceTopology) bool {
+			_, err = wait.With(fxt.Client).Interval(5*time.Second).Timeout(1*time.Minute).ForNodeResourceTopologiesEqualTo(ctx, &nrtInitialList, func(nrt *nrtv1alpha2.NodeResourceTopology) bool {
 				return !nodesNameSet.Has(nrt.Name)
 			})
 			Expect(err).ToNot(HaveOccurred())
 
-			By("updating deployment in such way that some pods will fit into NUMA nodes")
-			err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), dp)
-			Expect(err).ToNot(HaveOccurred())
+			// we should expect 'expectedReadyReplicas' out of total replica pods to be running
+			// each NUMA cell should hold a single pod, so we should expect the number of replicas to be equal to number of available NUMAs
+			// multiNUMACandidates nodes has 2 NUMAs each
+			expectedReadyReplicas := int32(nrtCandidates * 2)
 
-			// 6 pods in total (replica is 6)
-			// we should expect 'expectedReadyReplicas' out of 6 pods to be running
-			expectedReadyReplicas := calcExpectedReadyReplicas(numOfnrtCandidates, len(nrts)-numOfnrtCandidates)
-			klog.Infof("expecting %d out of %d to be running", expectedReadyReplicas, replicas)
+			By("updating deployment in such way that some pods will fit into NUMA nodes")
+			Eventually(func() error {
+				err := fxt.Client.Get(ctx, client.ObjectKeyFromObject(dp), dp)
+				if err != nil {
+					return err
+				}
 
-			numaLevelFitRequiredRes := corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("2"),
-				corev1.ResourceMemory: resource.MustParse("1Gi"),
-			}
-			cnt := &dp.Spec.Template.Spec.Containers[0]
-			cnt.Resources.Limits = numaLevelFitRequiredRes
-			cnt.Resources.Requests = numaLevelFitRequiredRes
+				klog.Infof("expecting %d out of %d to be running", expectedReadyReplicas, replicas)
```

Comment:
this log will be printed out every second, I think it better be moved outside the eventually especially because here we only update the deployment but don't assert any condition about the replicas.

Reply:
yes, 60 times is too much. I'll add code to print less often, I want to keep some feedback mechanism during the update loop